### DISCUSSION TITLE
Can't get streams-i2s-webserver_wav to work, could it be the pins I use? If not then what?

### DISCUSSION THREAD

**Discussion by CataCluj** [pschatzmann/arduino-audio-tools#918] [Q&A] [answered]
Hi, and thank you for a great library.
Using latest Arduino IDE 2.1.1 with everything up to date.
I'm unable to get any sound from the browser although it would seem from the logs that it's working; I get many of these:
"[I] StreamCopy.h : 139 - StreamCopy::copy 1024 -> 1024 -> 1024 bytes - in 1 hops"
I made my connections before I knew about this project so my pins are configured like this (it works with Home Assistant):
``
  auto config = i2sStream.defaultConfig(RX_MODE);
  config.port_no = I2S_NUM_0;   //Cat!! this was not here; unsure what number it should be; tried 0,1
  config.pin_ws = 15; 
  config.pin_bck = 0;
  config.pin_data = 27;
  config.signal_type = Digital;
  config.i2s_format = I2S_PHILIPS_FORMAT; // I tried maybe all other formats
  config.sample_rate = 16000;
  config.channels = 1;  //Cat!! was 2
  config.bits_per_sample = 16;
  i2sStream.begin(config);
  Serial.println("I2S started");
  server.begin(i2sStream, config, &filler);
``
Apart from why is it not working (tried Vivaldi, Opera, Edge, Edge in Idternet Explorer mode, (on Windows), Chrome (on Android)), I wonder why the Clock pin I've seen in all other examples is not 0, 1 or 3 (I used 0) as the ESP32 Datasheet specifies those are the only 3 pins that can be used for I2S clocks.  I realise that "soft" I2S can be implemented but the hard I2S should be better... no?
Thank you

**Comment by CataCluj** [comment]
I'm sorry, I tried "Add Code" and I don't know why it's formatted all in one line; I'll try a few other things.

**Comment by pschatzmann** [comment, marked as answer]
https://github.com/pschatzmann/arduino-audio-tools/issues/893

**Comment by CataCluj** [comment]
That was quick; thank you.
I'm trying VLC but I get:
Unidentified codec:
VLC could not identify the audio or video codec
Or is there anything else I could try easily on windows?

**Comment by pschatzmann** [comment]
https://github.com/pschatzmann/arduino-audio-tools/wiki/It's-not-working
Not sure what ADC you are using, but maybe it does not support 16 bits properly...
I hope you did not ignore the [related documentation](https://github.com/pschatzmann/arduino-audio-tools/wiki/External-ADC)

**Comment by CataCluj** [comment]
I'm using the INMP441 microphone.
As you see; I changed the pins but they work with ESPHome.  Also, how can I2S work with SCK/BCK on pins other than 0, 1, 3?

**Comment by CataCluj** [comment]
I may try the [streams-i2s-serial] sketch, but that won't solve the fact that browsers don't seem to work and I don't know how to make VLC recognize the stream :-(

**Comment by pschatzmann** [comment]
I think you mix up the master clock with the bit block! The bit clock can by on any pin...
The master clock is not used by the  INMP441

**Comment by CataCluj** [comment]
I probably do... now that you mention it; but is there any reason for my pin configuration to not work? I have a Lolin Lite board.
config.pin_ws = 15;  
config.pin_bck = 0; 
config.pin_data = 27;

**Comment by CataCluj** [comment]
Is it possible that turning on I2S somehow takes over GPIO0 with Master Clock and it can't be used for bit clock? Would that be different in ESPHome somehow?

**Comment by CataCluj** [comment]
The reason I'm doing this is because functionality in Home Assistant works but not great and I can't find a way to listen to it, to determine level/noise, etc to see what I might need to improve.

**Comment by pschatzmann** [comment]
Unlikely: as you can see the following default is defined in AudioConfig.h
#define PIN_I2S_MCK -1

**Comment by CataCluj** [comment]
So, any other way to listen to it, as it would seem to be mostly working?

**Comment by CataCluj** [comment]
I got the Serial program to work and I can see the signal; it looks pretty good to me except the autoscaling which is of course an IDE thing; not the library. 
I had t change int32_t to int16_t to get nicer numbers.
I'll look for a way to make this stop: "[I] StreamCopy.h : 139 - StreamCopy::copy 1024 -> 1024 -> 1024 bytes - in 1 hops"

Also, it would be nice if there was a way to also send some scaling/streaming numbers (max-min) to prevent the autoscaling; I wasn't able to do that properly just by putting some PrintLn in the main loop; and so far I didn't want to mess with the library itself...
Anyway, this is useful already as I can see I don't have big noise.
Thank you again.

**Reply by LinusHeu** [reply]
> I'll look for a way to make this stop: "[I] StreamCopy.h : 139 - StreamCopy::copy 1024 -> 1024 -> 1024 bytes - in 1 hops"

Looks correct to me. Just set the Logger to something lower than Info, to disable this message. Like Warning or Error or turn it off completely (if your project works, and you don't need it) https://github.com/pschatzmann/arduino-audio-tools#logging

**Reply by CataCluj** [reply]
Thank you Linus,
It seems (from AudioConfig.h) that default is "Warning" I tried this at the top of my ".ino" file:
> #define LOG_LEVEL AudioLogger::Error  //Could be "Error", "Warning", "Info", "Debug"
> #include "AudioTools.h"

but I still get that line.
I even added (between "#include "AudioTools.h" and "#define LOG_LEVEL AudioLogger::Error"
> #define USE_AUDIO_LOGGING false

To no avail.
Am I doing it in the wrong place?

**Reply by LinusHeu** [reply]
You need to do that in your AudioConfig.h. (Look at the documentation how that file works.)
Though I currently just use `AudioLogger::instance().begin(Serial, AudioLogger::Error);` inside my void setup()

**Reply by CataCluj** [reply]
But I put it before I include the h file so it should be defined as it gets in.
I may try "your way".
Thank you

**Reply by CataCluj** [reply]
I'm a Dummy (about this we have confirmation ;-); I'm sure about many other things too). I didn't notice that the logging level was overridden in the line instantiating AudioLogger, until I tried to do it "your way".
Dummy Me!
Thank you Linus!
Now, if there only was a way to insert some fixed highs/lows to stop AutoScaling...
